titus-server-master/src/main/java/com/netflix/titus/master/scheduler/constraint/V3ZoneBalancedHardConstraintEvaluator.java

```diff
@@ -0,0 +1,73 @@
+package com.netflix.titus.master.scheduler.constraint;
+
+import java.util.Map;
+
+import com.netflix.fenzo.ConstraintEvaluator;
+import com.netflix.fenzo.TaskRequest;
+import com.netflix.fenzo.TaskTrackerState;
+import com.netflix.fenzo.VirtualMachineCurrentState;
+import com.netflix.titus.master.jobmanager.service.common.V3QueueableTask;
+import com.netflix.titus.master.scheduler.SchedulerUtils;
+
+public class V3ZoneBalancedHardConstraintEvaluator implements ConstraintEvaluator {
+
+    private static final Result VALID = new Result(true, null);
+    private static final Result INVALID = new Result(false, "Zone balancing constraints not met");
+    private static final Result NO_ZONE_ID = new Result(false, "Host without zone data");
+
+    private final int expectedValues;
+    private final String zoneAttributeName;
+    private final TaskCache taskCache;
+
+    protected V3ZoneBalancedHardConstraintEvaluator(TaskCache taskCache, int expectedValues, String zoneAttributeName) {
+        this.taskCache = taskCache;
+        this.expectedValues = expectedValues;
+        this.zoneAttributeName = zoneAttributeName;
+    }
+
+    @Override
+    public String getName() {
+        return "V3ZoneBalancedHardConstraintEvaluator";
```

Comment:
I prefer spaces in between each of the words like the others.

Reply:
I changed other to use camel case like here.